docs/user/openstack/deploy_sriov_workers.md

```diff
@@ -0,0 +1,356 @@
+
+```yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+    machine.openshift.io/cluster-api-machine-role: <node_role>
+    machine.openshift.io/cluster-api-machine-type: <node_role>
+  name: <infrastructure_ID>-<node_role>
+  namespace: openshift-machine-api
+spec:
+  replicas: <number_of_replicas>
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+        machine.openshift.io/cluster-api-machine-role: <node_role>
+        machine.openshift.io/cluster-api-machine-type: <node_role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+    spec:
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          cloudName: openstack
+          cloudsSecret:
+            name: openstack-cloud-credentials
+            namespace: openshift-machine-api
+          flavor: <nova_flavor>
+          image: <glance_image_name_or_location>
+          serverGroupID: <optional_UUID_of_server_group>
+          kind: OpenstackProviderSpec
+          networks:
+          - subnets:
+            - uuid: <machines_subnet_uuid>
+            - filter:
+                name: radio
+              vnicType: direct
+              portCount: 4
+              portSecurity: false
+              portTags:
+                - "sriov"
+                - "radio"
+            - filter:
+                name: uplink
+              vnicType: direct
+              portCount: 4
+              portSecurity: false
+              portTags:
+                - "sriov"
+                - "uplink"
+          primarySubnet: <machines_subnet_uuid>
+          securityGroups:
+          - filter: {}
+            name: <infrastructure_ID>-<node_role>
+          serverMetadata:
+            Name: <infrastructure_ID>-<node_role>
+            openshiftClusterID: <infrastructure_ID>
+          tags:
+          - openshiftClusterID=<infrastructure_ID>
+          trunk: false
+          userDataSecret:
+            name: <node_role>-user-data
+          availabilityZone: <optional_openstack_availability_zone>
+```
+
+Once you have finished editing your machineSet, upload it to your OpenShift cluster:
+
+```sh
+c create -f sriov_machineset.yaml
+```
+
+## Creating SR-IOV Compute Nodes in UPI
```

Comment:
I still think we should clearly separate the upi and ipi workflows. Maybe put them in different sections altogether.

Reply:
Yeah, if there is enough of a difference, then I am in favor of it. I am still a bit fuzzy about what the IPI workflow should look like right now though.